api/tables/upload.js

```diff
@@ -0,0 +1,169 @@
+import {
+  DynamoDBClient,
+  BatchWriteItemCommand,
+  GetItemCommand,
+  QueryCommand
+} from '@aws-sdk/client-dynamodb'
+import { marshall, unmarshall } from '@aws-sdk/util-dynamodb'
+
+// https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-dynamodb/classes/batchwriteitemcommand.html
+export const BATCH_MAX_SAFE_LIMIT = 25
+
+/**
+ * Abstraction layer to handle operations on Upload Table.
+ *
+ * @param {string} region
+ * @param {string} tableName
+ * @param {object} [options]
+ * @param {string} [options.endpoint]
+ * @returns {import('../service/types').UploadTable}
+ */
+export function createUploadTable (region, tableName, options = {}) {
+  const dynamoDb = new DynamoDBClient({
+    region,
+    endpoint: options.endpoint
+  })
+
+  return {
+    /**
+     * Check if the given data CID is bound to the uploader DID
+     *
+     * @param {string} uploaderDID
+     * @param {string} dataCID
+     */
+     exists: async (uploaderDID, dataCID) => {
+      const cmd = new GetItemCommand({
+        TableName: tableName,
+        Key: marshall({
+          uploaderDID,
+          dataCID,
+        }),
+        AttributesToGet: ['uploaderDID'],
+      })
+  
+      try {
+        const response = await dynamoDb.send(cmd)
+        return response?.Item !== undefined
+      } catch {
+        return false
+      }
+    },
+    /**
+     * Link an upload to an account
+     *
+     * @param {string} uploaderDID
+     * @param {import('../service/types').UploadItemInput} item
+     */
+    insert: async (uploaderDID, { dataCID, carCIDs }) => {
+      /** @type {import('../service/types').UploadItemOutput[]} */
+      const items = []
+      const batchItems = []
+
+      // Create returning array and array to batch
+      for (const carCID of carCIDs) {
+        const item = {
+          uploaderDID,
+          dataCID,
+          carCID,
+          uploadedAt: new Date().toISOString(),
+        }
+
+        items.push(item)
+        batchItems.push({
+          ...item,
+          // add sk property for Dynamo Key uniqueness
+          sk: `${item.dataCID}#${item.carCID}`
+        })
+      }
```

Comment:
pull up uploadedAt to create once so all definitely get the same timestamp, and this seems like a simple `map` so would be a little clearer as
```suggestion
      const uploadedAt = new Date().toISOString(),

      const items = carCIDs.map(carCID => ({
        uploaderDID,
        dataCID,
        carCID,
        uploadedAt
      })

      // items formatted for dynamodb
      const batchItems = items.map(item => ({
        ...item,
        // add sk property for Dynamo Key uniqueness
        sk: `${item.dataCID}#${item.carCID}`
      })
```